src/models/validation/courseForm.ts

```diff
@@ -0,0 +1,49 @@
+import { z } from 'zod'
+
+
+export const youtubeLinkRegex = /^(https?:\/\/)?(www\.)?(youtube\.com|youtu\.be)\/.+$/
+export const LessonSchema = z.object({
+  name: z.string(),
+  youtube_link: z.string(),
+})
+
+export const CourseSchema = z.object({
+  name: z
+    .string({ required_error: 'أدخل اسم الدورة' })
+    .min(5, 'يجب أن يحتوي اسم الدورة على 5 أحرف على الأقل'),
+  description: z
+    .string({ required_error: 'أدخل وصف الدورة' })
+    .min(10, 'يجب أن يحتوي وصف الدورة على 10 أحرف على الأقل'),
+  cost: z
+    .number({ required_error: 'يجب على ثمن الدروة ان يتكون من أرقام' })
+    .nullable()
+    .refine(
+      (val) => val === null || val >= 0,
```

Comment:
to compare with both null and undefined
```suggestion
      (val) => val == null || val >= 0,
```